frontend/src/tests/application-tests.js

```diff
@@ -1,25 +1,268 @@
+
+            // switch to the new user
+            resp = await apiPOST("/debug/active_user", newUser);
+            expect(resp).toHaveStatus("success");
+
+            // submit an application to the posting
+            // assume posting submission is tested (previous test)
+            const surveyjsSubmission = {
+                answers: {
+                    utorid: "greenb",
+                    first_name: "Green",
+                    last_name: "Bee",
+                    email: "testemail@utoronto.ca",
+                    phone: "6471111111",
+                    student_number: "1002345678",
+                    position_preferences: {
+                        [position.position_code]: HIGH_PREFERENCE,
+                    },
+                },
+            };
+            resp = await apiPOST(
+                `/public/postings/${posting.url_token}/submit`,
+                surveyjsSubmission,
+                true
+            );
+            expect(resp).toHaveStatus("success");
+
+            // switch back to default (i.e. with admin role) user
+            resp = await apiPOST("/debug/active_user", defaultUser);
+            expect(resp).toHaveStatus("success");
+
+            // check db if new applicant and application has been created
+            resp = await apiGET("/admin/applicants");
+
+            foundUser = resp.payload.find(
+                (user) => user.utorid === newUser.utorid
+            );
+            expect(foundUser).toBeDefined();
+
+            resp = await apiGET(`/admin/sessions/${session.id}/applications`);
+
+            //assumes that the applicant and posting are primary keys for an application
+            originalApplication = resp.payload.find(
+                (application) =>
+                    application.applicant_id === foundUser.id &&
+                    application.posting_id === posting.id
+            );
+            expect(originalApplication).toBeDefined();
+        });
         it.todo(
-            "When submitting survey.js data an applicant and application are automatically created they don't exist"
-        );
-        it.todo(
-            "When submitting survey.js data an applicant and application are updated if they already exist"
+            "500 tests: do not link position preferences, and not sumit position"
         );
+        it("When submitting survey.js data an applicant and application are updated if they already exist", async () => {
+            // switch to the new user
+            resp = await apiPOST("/debug/active_user", newUser);
+            expect(resp).toHaveStatus("success");
+
+            // submit an updated application to the posting
+            const applicantInfo = {
+                utorid: "greenb",
+                first_name: "Not Green",
+                last_name: "And not Bee",
+                email: "testemail2@utoronto.ca",
+                phone: "6472111111",
+                student_number: "1012345678",
```

Comment:
Let's remove a couple of these fields so we can see if the old data is still preserved.